compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,298 @@
+}
+
+int entry(int argc, char **argv)
+{
+  // TODO Add new option names!
+
+  arser::Arser arser("circle-opselector provides selecting operations in circle model");
+
+  arser.add_argument("--version")
+    .nargs(0)
+    .required(false)
+    .default_value(false)
+    .help("Show version information and exit")
+    .exit_with(print_version);
+
+  // TODO Add new options!
+
+  arser.add_argument("input").nargs(1).type(arser::DataType::STR).help("Input circle model");
+  arser.add_argument("output").nargs(1).type(arser::DataType::STR).help("Output circle model");
+
+  // select option
+  arser.add_argument("--by_id")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Input operation id to select nodes.");
+  arser.add_argument("--by_name")
+    .nargs(1)
+    .type(arser::DataType::STR)
+    .help("Input operation name to select nodes.");
+
+  try
+  {
+    arser.parse(argc, argv);
+  }
+  catch (const std::runtime_error &err)
+  {
+    std::cerr << err.what() << std::endl;
+    std::cout << arser;
+    return EXIT_FAILURE;
+  }
+
+  std::string input_path = arser.get<std::string>("input");
+  std::string output_path = arser.get<std::string>("output");
+
+  std::string operator_input;
+
+  std::vector<int> by_id;
+  std::vector<std::string> by_name;
+
+  std::string op;
+  std::vector<int> oplist;
+  bool select_mode = false;
+
+  if (!arser["--by_id"] && !arser["--by_name"] || arser["--by_id"] && arser["--by_name"])
+  {
+    std::cerr << "Either option '--by_id' or '--by_name' must be specified" << std::endl;
+    std::cerr << arser;
+    return EXIT_FAILURE;
+  }
+
+  if (arser["--by_id"])
+  {
+    operator_input = arser.get<std::string>("--by_id");
+    split_id_input(operator_input, by_id);
+  }
+  if (arser["--by_name"])
+  {
+    operator_input = arser.get<std::string>("--by_name");
+    split_name_input(operator_input, by_name);
+  }
+
+  // option parsing test code.
+  for (int x : by_id)
+    std::cout << "by_id: " << x << std::endl;
+
+  for (std::string line : by_name)
+    std::cout << "by_name: " << line << std::endl;
+
+  // Load model from the file
+  foder::FileLoader file_loader{input_path};
+  std::vector<char> model_data = file_loader.load();
+
+  // Verify flatbuffers
+  flatbuffers::Verifier verifier{reinterpret_cast<uint8_t *>(model_data.data()), model_data.size()};
+  if (!circle::VerifyModelBuffer(verifier))
+  {
+    std::cerr << "ERROR: Invalid input file '" << input_path << "'" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  const circle::Model *circle_model = circle::GetModel(model_data.data());
+  if (circle_model == nullptr)
+  {
+    std::cerr << "ERROR: Failed to load circle '" << input_path << "'" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // Import from input Circle file
+  luci::Importer importer;
+  auto module = importer.importModule(circle_model);
+
+  // Select and Import from user input.
+  auto selector = std::make_unique<opselector::OpSelector>(circle_model);
```

Comment:
I think all `OpSelector` needs is just above imported `module` because `module` already has the information of circle model from flatbuffers. You don't have to start from the scratch.

But AFAIK, you don't have much time to re-implement the logic until due date. So, this is just for your information. So, if you think it is possible to re-implement it, please refer to the logic below. Otherwise, you can go with current implementation.

I would do like this.
```c++
// Get the circle nodes that user wants to select

// Check if the nodes are all connected

// Decide which node to sue for input and output
// You can use `loco::preds(node)` that returns node's input
// You can use `loco::succs(node)` that returns node's output

// Clone the nodes
// Please refer to 
// https://github.com/Samsung/ONE/blob/3e4a55187dbb789173986d526443e086b8d8e158/compiler/luci/partition/src/PartitionPModules.cpp#L94-L157
```

Reply:
Always appreciate your comment.

I will try re-implement it!